src/Widgets.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bolt;
+
+use Bolt\Snippet\QueueProcessor;
+use Bolt\Snippet\Target;
+use Bolt\Snippet\Zone;
+use Bolt\Widget\BaseWidget;
+use Bolt\Widget\BoltHeaderWidget;
+use Bolt\Widget\CanonicalLinkWidget;
+use Bolt\Widget\NewsWidget;
+use Bolt\Widget\SnippetWidget;
+use Bolt\Widget\TwigAware;
+use Bolt\Widget\WeatherWidget;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+use Tightenco\Collect\Support\Collection;
+use Twig\Environment;
+
+class Widgets
+{
+    /** @var Collection */
+    private $queue;
+
+    /** @var Request */
+    private $request;
+
+    /** @var QueueProcessor */
+    private $queueProcessor;
+
+    /** @var Environment */
+    private $twig;
+
+    public function __construct(RequestStack $requestStack, QueueProcessor $queueProcessor, Environment $twig)
+    {
+        $this->queue = new Collection([]);
+        $this->request = $requestStack->getCurrentRequest();
+        $this->queueProcessor = $queueProcessor;
+        $this->twig = $twig;
+    }
+
+    public function registerWidget(BaseWidget $widget): void
+    {
+        $widget->setRequest($this->request);
+
+        if ($widget instanceof TwigAware) {
+            $widget->setTwig($this->twig);
+        }
+
+        $this->queue->push([
```

Comment:
this should be an object, not array

Reply:
It's a `Collection`, this makes it easier to do things like "filter on target" and "sort by priority".

Reply:
I mean: what is pushed to queue should be an object, not array.

Reply:
I know, but this is easier to manage. by pushing arrays into the `Collection`, we can do filtering and sorting.

Reply:
you can do filtering and sorting of objects in collection...

Reply:
Changed. It now just adds `$widget` to the queue, and sorting/filtering is done on those directly.